### PR TITLE
DatabricksHook ClusterState

### LINKED ISSUE #19490: provider Databricks : add cluster API
### Description

Add Databricks Cluster API in Databricks Hook

### Use case/motivation

Databricks provider lacks [Cluster API](https://docs.databricks.com/dev-tools/api/latest/clusters.html) access to allow
to CRUD cluster as well as manage clusters (start/restart/delete/...)
Those API are very convenient to control clusters (get status, force shutdown...) and to optimize some specific flows (scale cluster in advance when a coming need is detected )

### Related issues

none

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Seokyun-Ha**
closes: https://github.com/apache/airflow/issues/19490

Organization: https://github.com/bagelcode-data
Co-workers: @Seokyun-Ha, @bskim45, @kyeonghoon-kim

Implement method for get Databricks Cluster State.

And, refactor `RunState` class also.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by Lee-W — APPROVED** [review]

File: tests/providers/databricks/hooks/test_databricks.py [hunk 1]
```diff
@@ -158,6 +162,11 @@ def repair_run_endpoint(host):
     """
     return f"https://{host}/api/2.1/jobs/runs/repair"
 
+def get_cluster_endpoint(host):
+    """
+    Utility function to generate the get run endpoint given the host.
+    """
+    return f"https://{host}/api/2.0/clusters/get"
```
Comment (frankcash) [inline tests/providers/databricks/hooks/test_databricks.py:170]: why not utilize the endpoint you defined in the hook?

Comment (phanikumv) [reply tests/providers/databricks/hooks/test_databricks.py:170]: I think the author wants to test whether the hook is returning the value correctly.

Comment (Seokyun-Ha) [reply tests/providers/databricks/hooks/test_databricks.py:170]: I just followed the pattern of exiting codes in the file 🙏

File: airflow/providers/databricks/hooks/databricks.py [hunk 2]
```diff
@@ -116,6 +118,55 @@ def from_json(cls, data: str) -> RunState:
...
+                "guide for troubleshooting information"
+            )
+
+        self.state = state
+        self.state_message = state_message
+
+    @property
+    def is_terminal(self) -> bool:
+        """True if the current state is a terminal state."""
+        return self.state in ("TERMINATING", "TERMINATED", "ERROR", "UNKNOWN")
```
Comment (utkarsharma2) [inline airflow/providers/databricks/hooks/databricks.py:149]: Not a blocker, you can probably also do something like this to not redefine the states in function  `is_running` and `is_terminal`
```
CLUSTER_TERMINAL_STATE = ["TERMINATING", "TERMINATED", "ERROR", "UNKNOWN"]
CLUSTER_RUNNING_STATE = ["RUNNING", "RESIZING"]
CLUSTER_LIFE_CYCLE_STATES = [
        "PENDING",
        "RESTARTING",
        *CLUSTER_RUNNING_STATE,
        *CLUSTER_TERMINAL_STATE
    ]
```
and then use CLUSTER_RUNNING_STATE in `is_running` and CLUSTER_TERMINAL_STATE in `is_terminal`

**Review by utkarsharma2 — APPROVED** [review]

File: airflow/providers/databricks/hooks/databricks.py (hunk 2 above)
Comment (Seokyun-Ha) [reply airflow/providers/databricks/hooks/databricks.py:149]: kind of good suggestion! 👍  However, for this PR, I will match the pattern with existing `RunState` class. Later, I will refactor both classes after this task 🙏

**Comment by Seokyun-Ha** [comment]
Hello, @alexott. We implemented Databricks GET cluster method for the hook. Please, take a look at 😄

**Review by alexott — APPROVED** [review]
lgtm, thank you!

**Comment by Seokyun-Ha** [comment]
Thanks you all guys to review this PR ❤️  I will request to schedule PR merge on Airflow slack. 

cc. @alexott @Lee-W @utkarsharma2

**Review by eladkal — APPROVED** [review]

**Pull request merged**